pkg/cloud/ovirt/providerIDcontroller/providerIDController.go

```diff
@@ -0,0 +1,146 @@
+package providerIDcontroller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	ovirtsdk "github.com/ovirt/go-ovirt"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/klogr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/openshift/cluster-api-provider-ovirt/pkg/cloud/ovirt/clients"
+)
+
+var _ reconcile.Reconciler = &providerIDReconciler{}
+
+type providerIDReconciler struct {
+	log                  logr.Logger
+	client               client.Client
+	listNodesByFieldFunc func(key, value string) ([]corev1.Node, error)
+	fetchProviderIDFunc  func(string) (string, error)
+	ovirtApi             *ovirtsdk.Connection
+}
+
+func (r *providerIDReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log := r.log.WithValues("node", request.NamespacedName)
+	log.V(3).Info("Reconciling", request.NamespacedName)
+
+	// Fetch the Node instance
+	node := corev1.Node{}
+	err := r.client.Get(context.Background(), request.NamespacedName, &node)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, fmt.Errorf("error getting node: %v", err)
+	}
+
+	if node.Spec.ProviderID != "" {
+		return reconcile.Result{}, nil
+	}
+
+	r.log.Info("spec.ProviderID is empty, fetching from ovirt")
```

Comment:
Need to use the child logger you created otherwise you won't get the values you've set. There are other places that have this issue as well.
```suggestion
	log.Info("spec.ProviderID is empty, fetching from ovirt")
```

I think the reason it doesn't modify `r.log` is so that you could reconcile multiple times concurrently, if each overwrote the values for `r.log` your logs would all get mangled as they're all fighting over who's `node` value should be set

Reply:
> Need to use the child logger you created otherwise you won't get the values you've set. There are other places that have this issue as well.
> 
> I think the reason it doesn't modify `r.log` is so that you could reconcile multiple times concurrently, if each overwrote the values for `r.log` your logs would all get mangled as they're all fighting over who's `node` value should be set

The reason I don't like this is because its really easy to misuse - like here I used 2 different log instances.

Reply:
I just gave up on the idea of extra log instance, simply doesn't worth the confusion.